pytest_plugins/rerun_rp/rerun_rp.py

```diff
@@ -113,13 +116,20 @@ def pytest_collection_modifyitems(items, config):
     fail_args = config.getoption('only_failed', False)
     skip_arg = config.getoption('only_skipped', False)
     user_arg = config.getoption('user', False)
-    if not any([fail_args, skip_arg, user_arg]):
+    upgrades_rerun = config.getoption('upgrades_rerun', False)
+    if not any([fail_args, skip_arg, user_arg, upgrades_rerun]):
         return
     rp = ReportPortal()
     version = settings.server.version
     sat_version = f'{version.base_version}.{version.epoch}'
     LOGGER.info(f'Fetching Report Portal launches for target Satellite version: {sat_version}')
-    launch = next(iter(rp.launches(sat_version=sat_version).values()))
+    launch = next(
+        iter(
+            rp.launches(
+                sat_version=sat_version, launch_type='upgrades' if upgrades_rerun else 'satellite6'
```

Comment:
This hardcoding (and use of) `'satellite6'` is something we need to move away from. The launch type should be pulled from report portal configuration, or directly passable.

Reply:
@mshriver by configuration you mean from current robottelo.properties (future dynaconf) ...  that should be doable and need since we are moving to DynaConf.

Reply:
We have that already available in RP, but thats basically for Report Portal Module. For plugin I just wanted to make it easier by `--upgrades_rerun` option.